lib/vsc/utils/fancylogger.py

```diff
@@ -562,6 +583,30 @@ def _logToSomething(handlerclass, handleropts, loggeroption, enable=True, name=N
     return handler
 
 
+def _screenLogFormatterFactory(colorize='never', stream=sys.stdout):
+    """
+    Return a log formatter class, with optional colorization features.
+
+    Second argument `colorize` controls whether the formatter
+    can use ANSI terminal escape sequences:
+
+    * ``'never'`` (default) forces use the plain `logging.Formatter` class;
+    * ``'always'`` forces use of the colorizing formatter;
+    * ``'auto'`` selects the colorizing formatter depending on whether `stream` is connected to a terminal.
+
+    Second argument `stream` is the stream to check in case `colorize` is ``'auto'``.
+    """
+    formatter = logging.Formatter  # default
+    if HAVE_COLOREDLOGS_MODULE:
+        if colorize == 'auto':
+            # auto-detect
+            if humanfriendly.terminal.terminal_supports_colors(stream):
+                formatter = coloredlogs.ColoredFormatter
+        elif colorize == 'always':
+            formatter = coloredlogs.ColoredFormatter
```

Comment:
do we care about other values than `auto`, `always` or `never` being passed down?

if so, this is missing:

```python
elif colorize != 'never':
    # FAIL
```